src/Xtools/ArticleInfo.php

```diff
@@ -334,7 +362,59 @@ public function topTenPercentage()
      */
     public function getPageviews($latest)
     {
-        return $this->page->getLastPageviews($latest);
+        if (false === $this->startDate && false === $this->endDate) {
+            return $this->page->getLastPageviews($latest);
+        }
+
+        list($start, $end) = $this->translateDatesToYYYYMMDD($this->startDate, $this->endDate);
+        list($start, $end) = $this->applyDatesDefaults($start, $end, $latest);
```

Comment:
I'm confused why we're using `$latest` in this case. The pageviews should be for the same date as what was requested. So if they only gave an end date, it'd use the date of the first revision to the specified end date.

Reply:
That was what I meant when I told you on IRC that I have minor logical-thinking problems :D